src/power_grid_model/typing.py

```diff
@@ -5,8 +5,8 @@
 """
 Type hints for PGM. This includes all miscellaneous type hints not under dataset or dataset_definitions categories
 """
-from power_grid_model.core.dataset_definitions import ComponentType
+from power_grid_model.core.dataset_definitions import ComponentTypeVar
 
-_ComponentAttributeMappingDict = dict[ComponentType, set[str] | list[str] | None]
+_ComponentAttributeMappingDict = dict[ComponentTypeVar, set[str] | list[str] | None]
```

Comment:
`ComponentAttributeMapping` can ingest a `ComponentAttributeMappingDict`. I don't think this is internal use only with the whole columnar thing. As a result, this should be `ComponentTypeVar`

unless we decide that we don't support `str` at all to users (we can still decide now because this is new functionality)

Reply:
Its marked as _. 
We convert `ComponentAttributeMapping` to `_ComponentAttributeMappingDict` internally and check if valid things are provided in `ComponentAttributeMapping`. And internally we can cast the string to ComponentType.

The users would not have any use for it. Either way, its really minor so you can resolve the comment.